core/src/main/java/com/linecorp/armeria/server/DefaultServiceRequestContext.java

```diff
@@ -174,8 +180,9 @@ private DefaultServiceRequestContext(
         this.routingContext = routingContext;
         this.routingResult = routingResult;
         this.sslSession = sslSession;
-        this.proxiedAddresses = proxiedAddresses;
-        this.clientAddress = requireNonNull(clientAddress, "clientAddress");
+        this.proxiedAddresses = requireNonNull(proxiedAddresses, "proxiedAddresses");
+        this.clientAddressMapper = requireNonNull(clientAddressMapper, "clientAddressMapper");
```

Comment:
How about removing `clientAddressMapper`? I think a user can still get it from `ServerConfig`, so it's better keeping request-scope members only for reduced memory footprint.